packages/core/src/components/table/table-body-cell/table-body-cell.scss

```diff
@@ -16,7 +16,7 @@
 }
 
 :host(.tds-table__body-cell--no-padding) {
-  padding: 0;
+  padding: 0 !important;
```

Comment:
I'm not a super fan of using !important. Without checking a substitute here, is there another way?

Reply:
Me neither. The alternative could be to use if-statements of some kind to enable conditional rendering of the classes, but since the classes are established in different files I'm not sure how straight forward that would be. 'compact-design' is set in the header and then propagates down to the body cells while 'disable-padding' is set straight in the body cells. Maybe there's a way to connect the two files? In my mind Table header would have to check the Table body cell to see if 'disable-padding' is present and then conditionally decide if the 'compact-design' class should be added or not, but maybe I'm making it more complicated than it needs to be?

Reply:
@nathalielindqvist maybe you can try and do this instead:

```
render() {
    const dynamicStyles = {
      textAlign: this.textAlignState,
      // Conditionally set padding style
      padding: this.disablePadding ? '0' : 'var(--tds-spacing-element-16)',
    };

    
    return (
      <Host
        class={{
          'tds-table__body-cell': true,
          'tds-table__body-cell--hover': this.activeSorting,
          'tds-table__compact': this.compactDesign,
          'tds-table--divider': this.verticalDividers,
          'tds-table--no-min-width': this.noMinWidth,
        }}
        style={dynamicStyles}
      >
        {this.cellValue}
        <slot />
      </Host>
    );
  }
```
I believe you can then remove the css for padding.

Reply:
It absolutely works for just toggling the correct padding for disablePadding, but if compactDesign and disablePadding are enabled at the same time the inline style overwrites the rules for the compactDesign and sets 16px when top and bottom should be 8px. It gets a little tricky when the props need to play ball with each other.

Reply:
I agree with Tim, lets try to avoid !important and find another way to resolve this.

Reply:
@nathalielindqvist ok, but if we do something like this:

```
let paddingStyle = 'var(--tds-spacing-element-16)'; // Default padding

    if (this.disablePadding) {
        paddingStyle = '0';
    } else if (this.compactDesign) {
        paddingStyle = 'var(--tds-spacing-element-8) var(--tds-spacing-element-16)';
    }
    
    const dynamicStyles = {
      textAlign: this.textAlignState,
      // Conditionally set padding style
      padding: paddingStyle
    };
```

Reply:
Thank you for the discussion! I removed the !important and implemented the conditional solution @timrombergjakobsson provided, thank you for that! Changes pushed :)

Reply:
Glad to help :)!